experimental/generation/generator/packages/library/src/dialogGenerator.ts

```diff
@@ -122,6 +125,24 @@ export async function writeFile(path: string, val: string, feedback: Feedback, s
     }
 }
 
+// Return template directories by combining explicit ones with library ones
+export async function templateDirectories(templateDirs?: string[]): Promise<string[]> {
+    // Fully expand all directories
+    templateDirs = resolveDir(templateDirs || [])
+
+    // User templates + cli templates to find schemas
+    let startDirs = [...templateDirs]
```

Comment:
Nit: Prefer to use const when the reference doesn't change (even if the value changes)